shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/WebClientPlugin.java

```diff
@@ -73,7 +76,13 @@ public WebClientPlugin(final WebClient webClient) {
         int retryTimes = (int) Optional.ofNullable(exchange.getAttribute(Constants.HTTP_RETRY)).orElse(0);
         log.info("The request urlPath is {}, retryTimes is {}", urlPath, retryTimes);
         HttpMethod method = HttpMethod.valueOf(exchange.getRequest().getMethodValue());
-        WebClient.RequestBodySpec requestBodySpec = webClient.method(method).uri(urlPath);
+        try {
+            uriObject = new URI(urlPath);
+        } catch (Exception e) {
+            log.error("build URI object error",e);
+            return WebFluxResultUtils.result(exchange, e);
+        }
+        WebClient.RequestBodySpec requestBodySpec = webClient.method(method).uri(uriObject);
```

Comment:
Please use method `java.net.URI#create`

Reply:
Has been modified according to your suggestion

Reply:
WebClient.RequestBodySpec requestBodySpec = webClient.method(method).uri(URI.create(urlPath));

Reply:
ok, has been modified according to your suggestion